src/overlays/actors/ovl_En_Kakasi2/z_en_kakasi2.c

```diff
@@ -20,21 +42,193 @@ const ActorInit En_Kakasi2_InitVars = {
+
+    SkelAnime_ChangeAnim(&this->skelAnime, &D_06000214, 1.0f, 0.0f, (s16)frameCount, 0, -10.0f);
+    Audio_PlayActorSound2(&this->actor, NA_SE_EV_COME_UP_DEKU_JR);
+    this->actionFunc = func_80A90578;
+}
+
+void func_80A90578(EnKakasi2* this, GlobalContext* globalCtx) {
+    s16 currentFrame;
+
+    SkelAnime_FrameUpdateMatrix(&this->skelAnime);
+
+    currentFrame = this->skelAnime.animCurrentFrame;
+    if (currentFrame == 11 || currentFrame == 17) {
+        Audio_PlayActorSound2(&this->actor, NA_SE_EV_KAKASHI_SWING);
+    }
+
+    this->actor.shape.rot.y += 0x800;
+    Math_SmoothDownscaleMaxF(&this->actor.shape.unk_08, 0.5f, 500.0f);
+
+    if (this->actor.shape.unk_08 > -100.0f) {
+        this->actionFunc = func_80A9062C;
+        this->actor.shape.unk_08 = 0.0f;
+    }
+}
+
+void func_80A9062C(EnKakasi2* this, GlobalContext* globalCtx) {
+    f32 frameCount = SkelAnime_GetFrameCount(&D_06000214.genericHeader);
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Kakasi2/func_80A90264.s")
+    SkelAnime_ChangeAnim(&this->skelAnime, &D_06000214, 0.0f, 0.0f, (s16)frameCount, 2, -10.0f);
+    this->actionFunc = func_80A906C4;
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Kakasi2/func_80A904D8.s")
+void func_80A906C4(EnKakasi2* this, GlobalContext* globalCtx) {
+    if (this->skelAnime.animCurrentFrame != 0) {
+        Math_SmoothDownscaleMaxF(&this->skelAnime.animCurrentFrame, 0.5f, 1.0f);
+    }
+    Math_SmoothScaleMaxMinS(&this->actor.shape.rot.y, this->unk_198, 0x5, 0xBB8, 0x0);
+    SkelAnime_FrameUpdateMatrix(&this->skelAnime);
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Kakasi2/func_80A90578.s")
+void EnKakasi2_Update(Actor* thisx, GlobalContext* globalCtx) {
+    EnKakasi2* this = THIS;
+    GlobalContext* globalCtx2 = globalCtx;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Kakasi2/func_80A9062C.s")
+    this->actor.posRot.rot = this->actor.shape.rot;
+    Actor_SetHeight(&this->actor, this->height);
+    this->actionFunc(this, globalCtx2);
+    Actor_MoveForward(&this->actor);
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Kakasi2/func_80A906C4.s")
+    if (this->actor.shape.unk_08 == 0.0f) {
+        Collider_CylinderUpdate(&this->actor, &this->collider);
+        CollisionCheck_SetAC(globalCtx2, &globalCtx2->colChkCtx, &this->collider.base);
+        CollisionCheck_SetOC(globalCtx2, &globalCtx2->colChkCtx, &this->collider.base);
+    }
+    if (BREG(0) != 0) {
+        if (BREG(5) != 0) {
+            osSyncPrintf(VT_FGCOL(YELLOW) "☆☆☆☆☆ this->actor.player_distance ☆☆☆☆☆ %f\n" VT_RST, this->actor.xzDistFromLink);
+            osSyncPrintf(VT_FGCOL(YELLOW) "☆☆☆☆☆ this->hosei.x ☆☆☆☆☆ %f\n" VT_RST, this->distance.x);
+            osSyncPrintf("\n\n");
+        }
+        if (this->actor.draw == NULL) {
+            if (this->unk_194 != 0) {
+                if (!(this->unk_194 & 1)) {
```

Comment:
```suggestion
                if ((this->unk_194 & 1) == 0) {
```
`unk_194` looks like a timer
even better if modulus works:
```suggestion
                if ((this->unk_194 % 2) == 0) {
```